app/controllers/devise/registrations_controller.rb

```diff
@@ -63,11 +63,16 @@ def update
 
   # DELETE /resource
   def destroy
-    resource.destroy
-    Devise.sign_out_all_scopes ? sign_out : sign_out(resource_name)
-    set_flash_message! :notice, :destroyed
-    yield resource if block_given?
-    respond_with_navigational(resource){ redirect_to after_sign_out_path_for(resource_name) }
+    if resource.class.require_password_to_destroy
+      if destroy_resource(resource, account_destroy_params)
+        sign_out_resource(resource_name, resource)
+      else
+        render :edit
+      end
+    else
+      resource.destroy
+      sign_out_resource(resource_name, resource)
+    end
```

Comment:
Maybe:

```ruby
if destroy_resource(resource, account_destroy_params)
  sign_out_resource(resource_name, resource)
else
  render :edit
end
```

And then on destroy resource:

```ruby
def destroy_resource(resource, params)
  if resource.class.require_password_to_destroy
    resource.destroy_with_password(params)
  else
    resource.destroy
  end
end
```

Reply:
Yeah I agree, that's way better. I will do these changes.